src/core/layout.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {devAssertElement} from './assert';
+
+/**
+ * Returns "true" for internal AMP nodes or for placeholder elements.
+ * @param {!Node} node
+ * @return {boolean}
+ */
+export function isInternalOrServiceNode(node) {
+  if (node.nodeType !== Node.ELEMENT_NODE) {
+    return false;
+  }
+  node = devAssertElement(node);
+
+  if (node.tagName.toLowerCase().startsWith('i-')) {
+    return true;
+  }
+  if (
+    node.hasAttribute('placeholder') ||
+    node.hasAttribute('fallback') ||
+    node.hasAttribute('overflow')
+  ) {
+    return true;
+  }
+  return false;
+}
```

Comment:
Yes! But that requires changing 100s of files import paths and so I'd rather do it separately since my main goal is to unblock https://github.com/ampproject/amphtml/pull/34770

Reply:
If I understand correctly, this is only used in custom-element. What's the motivation for moving it into core, rather than closer to CE?

Reply:
Compiler will need to run it, and we don't want compiler to need an instance of `CustomElement\